clusterloader2/docs/GETTING_STARTED.md

```diff
@@ -0,0 +1,303 @@
+# ClusterLoader2
+
+In this tutorial, we will:
+- Set-up perf-tests repository for local development
+- Create single node cluster using [Kind]
+- Implement a simple CL2 test and run it
+- Run load test on 100 nodes cluster
+
+## Clone perf-tests repository
+
+Start with cloning perf-tests repository:
+```bash
+git clone git@github.com:kubernetes/perf-tests.git
+cd perf-tests
+```
+
+## Install GVM
+Follow instructions on [GVM install].
+Install golang with specific version (1.15.12 was tested in this tutorial):
+```bash
+gvm install go1.15.12
+gvm use go1.15.12
+```
+Next, add perf-tests repository to GOPATH:
+
+```bash
+gvm linkthis k8s.io/perf-tests
+```
+
+## Create cluster using kind
+Follow the [kind installation][Kind install] guide.
+
+Create cluster v1.21.1 with one master and one node:
+```bash
+kind create cluster --image=kindest/node:v1.21.1 --name=test-cluster --wait=5m
+```
+
+This command additionally generates cluster access credentials which are stored in `${HOME}/.kube/config` within a context named test-cluster.
+
+Check that you can connect to cluster:
+```bash
+kubectl get nodes
+```
+
+
+## Prepare simple test to run
+Let's prepare our first test config (config.yaml).
+This test will:
+- Create one namespace
+- Create a single deployment with 10 pods inside that namespace
+- Measure startup latency of these pod
+
+We will create file `config.yaml` that describes this test.
+First we need to start with defining test name:
+```yaml
+name: test
+```
+CL2 will create namespaces automatically, but we need to specify how many namespaces we want:
+```yaml
+namespace:
+  number: 1
+```
+Next, we need to specify TuningSets.
+TuningSet describes how actions are executed.
+In our case, we will have only 1 deployment so there will be only 1 action to execute.
+```yaml
+tuningSets:
+- name: Uniform1qps
+  qpsLoad:
+    qps: 1
+```
+Test definition consists of a list of steps. A step can be either collection of Phases or Measurements.
+A Phase defines a state the cluster should reach.
+A Measurement allows to measure something or wait for something.
+You can find list of available measurements here [Measurements].
+Our first step will be starting two measurements.
+We want to start measuring pod startup latency and also make measurement that will wait for all pods to be in running state.
+Setting the action field to `start` begins execution of measurement.
+For both measurements we need to specify labelSelectors so they know which pods they should take into account.
+PodStartupLatency also takes threshold. If 99th percentile of latency will go over this threshold, test will fail.
```

Comment:
Let's remove - it has a default and using that in the first place isn't necessary the most important thing...